tfexec/errors.go

```diff
@@ -0,0 +1,74 @@
+package tfexec
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func parseError(stderr string) error {
+	switch {
+	// case ErrTerraformNotFound.regexp.MatchString(stderr):
+	// return ErrTerraformNotFound
+	case regexp.MustCompile(usageRegexp).MatchString(stderr):
+		return &ErrCLIUsage{stderr: stderr}
+	case regexp.MustCompile(`Error: Could not satisfy plugin requirements`).MatchString(stderr):
+		return &ErrNoInit{stderr: stderr}
+	case regexp.MustCompile(`Error: No configuration files`).MatchString(stderr):
+		return &ErrNoConfig{stderr: stderr}
+	default:
+		return &Err{stderr: stderr}
+	}
+}
+
+type ErrNoSuitableBinary struct {
```

Comment:
Errors should probably be suffixed with `...Error` to mimic other Go conventions:

* https://pkg.go.dev/os?tab=doc
* https://pkg.go.dev/net/http?tab=doc#ProtocolError

etc...